### DISCUSSION TITLE
[ 草案规划 ] 插件安全与可信度分级

### DISCUSSION THREAD

**Discussion by SOVLOOKUP** [rubickCenter/rubick#56] [Ideas]
插件化方案虽然有极大的拓展性和自由度，但实际上是在本地运行不可信代码，有一定的安全隐患。比如 xtools 的插件权限无限制，也没有系统安全与隐私保护方案，这是我不敢用它的原因之一。

我想无论是 Chrome 插件 (Manifest V3) 还是 PWA，他们不敢过多开放原生能力的原因就是安全和隐私的问题。 

当以后社区插件越来越多的时候，如何保证 rubick 的使用安全，保证用户不会使用到恶意插件呢？

我想可以通过审核给插件一个可信度分级，然后 rubick 在运行时根据插件可信度级别赋予其相应的 API 权限，如下表：

| 插件可信度 \ API 权限 | CPU,内存,数据,用户等 | 网络请求 | 文件读写 | 系统控制 |
| --------------------- | -------------------- | -------- | -------- | -------- |
| 未认证 不开源         | ✔️                    | ❌        | ❌        | ❌        |
| 社区认证 不开源       | ✔️                    | ✔️        | ❌        | ❌        |
| 社区认证 开源         | ✔️                    | ✔️        | ✔️        | ❌        |
| 官方认证              | ✔️                    | ✔️        | ✔️        | ✔️        |

**Comment by renmu123** [comment]
我觉得没必要限制这么严格，只分为两种：
1. 官方认证审核，允许直接在应用搜索安装
2. 其余只能手动进行安装

就算 vscode 的插件也做不到你说得这种程度。
也无法确保在通过认证后进行作恶。

我们只能确保在商店中的插件是正常的，用户自行安装的只能默认用户对这个插件具有信任。
